src/maplibre_helpers.ts

```diff
@@ -60,12 +64,16 @@ export function drawCircle(
 // layer. This complicates Vite's hot-reload feature, unless every component
 // correctly tears down all sources and layers. These methods workaround that
 // lifetime management hassle by overwriting if necessary.
-export function overwriteSource(map, id, source) {
+export function overwriteSource(
+  map: Map,
+  id: string,
+  source: SourceSpecification
+) {
   if (map.getSource(id)) {
     // First remove all layers using this source
     let layers = [];
     for (let layer of map.getStyle().layers) {
-      if (layer.source == id) {
+      if ("source" in layer && layer.source == id) {
```

Comment:
`LayerSpecification` has a bunch of union cases, and one of them has no `source`. Apparently this is the canonical way to deal